Add fault diagram to ASB, RabbitMQ, and SQS topology diagram that explains the flow of messages for errors (#5501)
MassTransit/MassTransit: doc/content/3.documentation/6.transports/4.amazon-sqs.md

SNS also provides a feature called fan-out delivery, which enables messages to be delivered to multiple subscribers in parallel, this allows SNS to handle high-throughput and burst traffic, and can improve the overall performance of your application.

MassTransit uses SNS to route published messages to SQS queues.

## Broker Topology

> In this example topology the following messages are used:

Reviewer: ```suggestion
The following messages are used in this example:
```

> The command contract for processing a file that was received.

Reviewer: Making this align with the ASB text

```suggestion
Here is the command contract for processing a file that was received:
```

```csharp
namespace Acme;

public record ProcessFile
{
}
```

> The *event* contract that is published when the file was succesfully processed:

Reviewer: Making this align with the ASB text

```suggestion
These are the event contracts for a consumer that receives files from a customer:
```

```csharp
namespace Acme;

public record FileReceivedEvent
{
}
```

> The following consumers for these message contracts are as below.

Reviewer: Making this align with the ASB text

```suggestion
The consumers for these message contracts are shown below:
```

```csharp
class ProcessFileConsumer :
    IConsumer<ProcessFile>
{
}

class FileReceivedConsumer :
    IConsumer<FileReceivedEvent>
{
}

class CustomerAuditConsumer :
    IConsumer<FileReceivedEvent>
{
}
```

:::alert{type="info"}
The broker topology can be customized using the [topology API](/documentation/configuration/topology).
:::

### Send

> These are the exchanges and queues for above mentioned example topology for a Send:

Reviewer: Making this align with the ASB text

```suggestion
These are the exchanges and queues for the example above when Sending a message:
```

![Send topology for Azure Service Bus](/amazonsqs-topology-send.svg)

### Publish

> These are the exchanges and queues for above mentioned example topology for a Publish:

Reviewer: Making this align with the ASB text

```suggestion
These are the topics and queues for the example above when Publishing a message:
```

![Publish topology for Azure Service Bus](/amazonsqs-topology-publish.svg)

### Fault

> These are the exchanges and queues used when messages fail. The failing message gets forwarded to an `_error` queue by default. The following diagram shows which Exchanges and Queues are used when a message fails to be processed and is deadlettered for the above mentioned example topology, as well as how fault result messages are sent.

Reviewer: ```suggestion
These are the exchanges and queues used when messages fail. The failing message gets forwarded to an `_error` queue by default. The following diagram shows which exchanges and queues are used when a message fails to be processed and is deadlettered for the example above.
```
